openedx/features/marketplace/views.py

```diff
@@ -1,24 +1,145 @@
+    MARKETPLACE_PARAM_COUNTRY_KEY,
+    MARKETPLACE_PARAM_QUERY_KEY,
+    ORGANIZATIONAL_PROBLEM_CHOICES_VIEW,
+    ORGANIZATIONAL_PROBLEM_TYPE_DELIVERY_SERVICES_KEY,
+    ORGANIZATIONAL_PROBLEM_TYPE_FUNDING_KEY,
+    ORGANIZATIONAL_PROBLEM_TYPE_HEALTH_CARE_SUPPLIES_KEY,
+    ORGANIZATIONAL_PROBLEM_TYPE_HUMAN_RESOURCES_KEY,
+    ORGANIZATIONAL_PROBLEM_TYPE_MENTORSHIP_KEY,
+    ORGANIZATIONAL_PROBLEM_TYPE_ONLINE_TRAINING_KEY,
+    ORGANIZATIONAL_PROBLEM_TYPE_OTHER_KEY,
+    ORGANIZATIONAL_PROBLEM_TYPE_REMOTE_WORKING_TOOLS_KEY,
+    ORGANIZATIONAL_SECTOR_ART_AND_CULTURE,
+    ORGANIZATIONAL_SECTOR_CLEAN_ENERGY,
+    ORGANIZATIONAL_SECTOR_CLIMATE_CHANGES,
+    ORGANIZATIONAL_SECTOR_EDUCATION,
+    ORGANIZATIONAL_SECTOR_ENVIRONMENTAL_CONSERVATION,
+    ORGANIZATIONAL_SECTOR_GENDER_EQUALITY,
+    ORGANIZATIONAL_SECTOR_HEALTH_AND_WELL_BEING,
+    ORGANIZATIONAL_SECTOR_HUMAN_RIGHTS,
+    ORGANIZATIONAL_SECTOR_OTHER_KEY,
+    ORGANIZATIONAL_SECTOR_SANITATION,
+    ORGANIZATIONAL_SECTOR_SOCIAL_JUSTICE,
+    ORGANIZATIONAL_SECTOR_WORK_AND_ECONOMIC_GROWTH,
+    ORGANIZATION_SECTOR_CHOICES_VIEW
+)
 from .models import MarketplaceRequest
 
 
-class MarketplaceListingView(View):
+class MarketplaceListView(ListView):
+    model = MarketplaceRequest
+    context_object_name = 'marketplace_list'
+    paginate_by = 10
     template_name = 'features/marketplace/marketplace_listing.html'
+    ordering = ['-created']
+    template_engine = 'mako'
 
-    def get(self, request, *args, **kwargs):
-        return render_to_response(self.template_name, {})
+    def get_checked_choices(self, choices):
+        """
+        This method takes list of tuples (choices) and returns a list
+        of choice keys that exist in self.request.GET
+        :param: choices
+        :return: list of choice keys that are also in self.request.GET
+        """
+        params = self.request.GET
+        tuple_key_index = 0
+        return [choice[tuple_key_index] for choice in choices if params.get(choice[tuple_key_index])]
+
+    def get_queryset(self):
+        params = self.request.GET
+
+        query = params.get(MARKETPLACE_PARAM_QUERY_KEY)
+        city = params.get(MARKETPLACE_PARAM_CITY_KEY)
+        country = params.get(MARKETPLACE_PARAM_COUNTRY_KEY)
+
+        request_type_list = self.get_checked_choices(ORGANIZATIONAL_PROBLEM_CHOICES_VIEW)
+        organizational_sector_list = self.get_checked_choices(ORGANIZATION_SECTOR_CHOICES_VIEW)
+        queryset = super(MarketplaceListView, self).get_queryset()
+
+        if request_type_list:
+            # has other problem selected
+            if ORGANIZATIONAL_PROBLEM_TYPE_OTHER_KEY in request_type_list:
+                queryset = queryset.filter(reduce(operator.or_, (~Q(organizational_problems__endswith='|')
+                                                                 | Q(organizational_problems__contains=x)
+                                                                 for x in request_type_list)))
```

Comment:
Please use a meaningful variable name instead of `x` here and in all the other places.

Reply:
Use appropriate name for this queryset that defines this queryset.